pyproject.toml

```diff
@@ -41,22 +33,81 @@ classifiers = [
     "Topic :: System :: Distributed Computing",
 ]
 
-[project.urls]
-Homepage = "https://github.com/locustio/locust"
-Documentation = "https://docs.locust.io/"
-Code = "https://github.com/locustio/locust"
-"Help/Questions" = "https://stackoverflow.com/questions/tagged/locust"
-"Issue tracker" = "https://github.com/locustio/locust/issues"
+packages = [{ include = "locust" }]
+include = [
+    "locust/webui/dist/**/*",
+    "locust/_version.py",
+    { path = "poetry.lock", format = [
+        "sdist",
+    ] },
+]
+exclude = [
+    "locust/webui/public",
+    "locust/webui/src",
+    "locust/webui/*.*",
+    "locust/test",
+]
 
+[tool.poetry.build]
+script = "build.py"
+generate-setup-file = false
 
-[project.readme]
-file = "README.md"
-content-type = "text/markdown"
+[tool.poetry.urls]
+"Help/Questions" = "https://stackoverflow.com/questions/tagged/locust"
+"Issue tracker" = "https://github.com/locustio/locust/issues"
 
-[tool.setuptools]
-zip-safe = false
-license-files = ["LICENSE"]
-include-package-data = false
+[tool.poetry-dynamic-versioning]
+enable = true
+vcs = "git"
+style = "pep440"
+format = "{base}{stage}"
+latest-tag = true
+pattern = "^(?P<base>\\d+\\.\\d+\\.\\d+)(?P<stage>.*)$"
+
+[tool.poetry-dynamic-versioning.files."locust/_version.py"]
+persistent-substitution = true
+initial-content = """
```

Comment:
Works with the current setup like-for-like, otherwise could be tweaked if the file can change